rnd/autogpt_server/autogpt_server/executor/manager.py

```diff
@@ -160,27 +173,67 @@ def register_next_execution(node_link: Link) -> NodeExecution | None:
                 )
             )
 
+        # Complete missing static input pins data using the last execution input.
+        static_link_names = {
+            link.sink_name
+            for link in next_node.input_links
+            if link.is_static and link.sink_name not in next_node_input
+        }
+        if static_link_names and (
+            latest_execution := wait(get_latest_execution(next_node_id, graph_exec_id))
+        ):
+            for name in static_link_names:
+                next_node_input[name] = latest_execution.input_data.get(name)
+
         next_node_input, validation_msg = validate_exec(next_node, next_node_input)
-        suffix = f"{next_output_name}>{next_input_name}~{next_node_id}:{validation_msg}"
+        suffix = (
+            f"{next_output_name}>{next_input_name}~{next_node_exec_id}:{validation_msg}"
+        )
 
         if not next_node_input:
             logger.warning(f"{prefix} Skipped queueing {suffix}")
-            return
+            return enqueued_executions
 
         # Input is complete, enqueue the execution.
         logger.warning(f"{prefix} Enqueued {suffix}")
-        execution_update(next_node_exec_id, ExecutionStatus.QUEUED)
-        return NodeExecution(
-            graph_exec_id=graph_exec_id,
-            node_exec_id=next_node_exec_id,
-            node_id=next_node.id,
-            data=next_node_input,
+        enqueued_executions.append(
+            add_enqueued_execution(next_node_exec_id, next_node_id, next_node_input)
         )
 
+        if not node_link.is_static:
+            return enqueued_executions
+
+        # If link is static, there could be some incomplete executions waiting for it.
```

Comment:
Main change 2:
Whenever an output is produced, if the link is static, we need to check if there are any incomplete executions that are still waiting on this input, because we can re-use the output as the input of these incomplete executions, and enqueue it on the execution queue.